dialogue-client-test-lib/src/main/java/com/palantir/dialogue/AbstractChannelTest.java

```diff
@@ -196,165 +192,122 @@ public void fillsQueryParameters() throws Exception {
+        ListenableFuture<Response> call = channel.createCall(endpoint, request);
+        assertThatCode(() -> call.cancel(true)).doesNotThrowAnyException();
+        assertThatCode(() -> call.cancel(true)).doesNotThrowAnyException();
     }
 
     @Test
     public void callCancellationIsObservedAsException() throws InterruptedException {
-        channel.createCall(endpoint, request).execute(mock(Observer.class));  // drain enqueued response
+        channel.createCall(endpoint, request);  // drain enqueued response
 
         channel = createChannel(server.url("").url(), Executors.newCachedThreadPool());
-        Call call = channel.createCall(endpoint, request);
-        call.execute(observer);
-        call.cancel();
+        ListenableFuture<Response> call = channel.createCall(endpoint, request);
+        Futures.addCallback(call, callback, MoreExecutors.directExecutor());
+        call.cancel(true);
 
         Thread.sleep(1000);
         server.enqueue(new MockResponse());
-        verify(observer, never()).success(any());
+        verify(callback, never()).onSuccess(any());
         ArgumentCaptor<Throwable> throwable = ArgumentCaptor.forClass(Throwable.class);
-        verify(observer).exception(throwable.capture());
+        verify(callback).onFailure(throwable.capture());
         assertThat(throwable.getValue())
                 // Different exceptions for HttpClient vs OkHttpClient
                 .isInstanceOfAny(CancellationException.class, IOException.class)
                 .hasMessageFindingMatch("(Task was cancelled\\.)|(Canceled)");
-        verify(observer, never()).failure(any());
     }
 
     // TODO(rfink): How to test that cancellation propagates to the server?
 
     @Test
     public void connectionErrorsSurfaceAsExceptions() throws IOException {
         server.shutdown();
-        channel.createCall(endpoint, request).execute(observer);
-        verify(observer, never()).success(any());
-        verify(observer).exception(any());
-        verify(observer, never()).failure(any());
-    }
-
-    @Test
-    public void conjureErrorsSurfaceAsFailures() throws IOException {
-        // drain successful response so we can enqueue a failed one below
-        channel.createCall(endpoint, request).execute(mock(Observer.class));
-
-        SerializableError error = SerializableError.forException(new ServiceException(ErrorType.FAILED_PRECONDITION));
-        server.enqueue(new MockResponse()
-                .setBody(new ObjectMapper().writeValueAsString(error))
-                .setResponseCode(500)
-                .addHeader("Content-Type", "application/json"));
-        channel.createCall(endpoint, request).execute(observer);
-        verify(observer, never()).success(any());
-        verify(observer, never()).exception(any());
-        verify(observer).failure(any());
-    }
-
-    @Test
-    public void unparseableNonConjureErrorsSurfaceExceptions() throws IOException {
```

Comment:
how do you think you will handle conjure vs application vs qos errors now?

Reply:
As above, I think you have something that maps the `Response` into something that differentiates an error from a conjure error you can allow callers to deal with that type. There's no need to deal with conjure errors down here, because nothing we're doing at the http layer here needs to know anything about conjure